shared/src/main/scala/tastyquery/ast/Symbols.scala

```diff
@@ -29,18 +29,35 @@ object Symbols {
     }
     final def tree: Tree = myTree
 
-    override def toString: String = s"symbol[$name]"
+    final def outer: Symbol = rawowner match {
+      case owner: Symbol => owner
+      case _             => assert(false, s"cannot access outer, $this was not declared within any scope")
```

Comment:
```suggestion
      case null          => assert(false, s"cannot access outer, $this was not declared within any scope")
```
so that we can trust that all cases are covered through exhaustivity checking. Same below.